core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -393,7 +394,9 @@ public void addColumn(
         checkDistressed();
 
         if (getColumnIndexQuiet(metaMem, name, columnCount) != -1) {
-            throw CairoException.instance(0).put("Duplicate column name: ").put(name);
+            sink.clear();
```

Comment:
It's an exceptional case, I'd use here `Misc.getThreadLocalBuilder()` instead of allocating an instance of `StringSink`